app/controllers/api/v1/registrant/contacts_controller.rb

```diff
@@ -62,7 +62,7 @@ def update
           disclosed_attributes = reparsed_request_json[:disclosed_attributes]
 
```

Comment:
very huge action comes here and there is a lot of checkers. I think it will be better to devide this action to separate methods and give them good names to understand what here happens